include/secp256k1_musig.h

```diff
@@ -375,113 +178,329 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
  *  encoded numbers are out of range, signature verification with it is
  *  guaranteed to fail for every message and public key.
  */
-SECP256K1_API int secp256k1_musig_partial_signature_parse(
+SECP256K1_API int secp256k1_musig_partial_sig_parse(
     const secp256k1_context* ctx,
-    secp256k1_musig_partial_signature* sig,
+    secp256k1_musig_partial_sig* sig,
     const unsigned char *in32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Computes an aggregate public key and uses it to initialize a keyagg_cache
+ *
+ *  Different orders of `pubkeys` result in different `agg_pk`s.
+ *
+ *  The pubkeys can be sorted before combining with `secp256k1_xonly_sort` which
+ *  ensures the same `agg_pk` result for the same multiset of pubkeys.
+ *  This is useful to do before `pubkey_agg`, such that the order of pubkeys
+ *  does not affect the aggregate public key.
+ *
+ *  Returns: 0 if the arguments are invalid, 1 otherwise
+ *  Args:        ctx: pointer to a context object initialized for verification
+ *           scratch: scratch space used to compute the aggregate pubkey by
+ *                    multiexponentiation. Generally, the larger the scratch
+ *                    space, the faster this function. However, the returns of
+ *                    providing a larger scratch space are diminishing. If NULL,
+ *                    an inefficient algorithm is used.
+ *  Out:      agg_pk: the MuSig-aggregated x-only public key. If you do not need it,
+ *                    this arg can be NULL.
+ *      keyagg_cache: if non-NULL, pointer to a musig_keyagg_cache struct that
+ *                    is required for signing (or verifying the MuSig protocol).
```

Comment:
```suggestion
 *                    is required for signing (or observing the signing session
 *                    and verifying partial signatures).
```

We could describe the aggregator node further in another PR.